main.go

```diff
@@ -153,6 +157,61 @@ func main() {
 			}
 		}()
 	} else {
+		if config.EnableProfiling {
```

Comment:
Shall we move this to another func like `func setupProfiling(logr *slog.Logger)`?

Reply:
Yah, let me organize the code better, lazily just pushed what I had written for profiling.